sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -1442,33 +1446,152 @@ ThroughputControlGroup enableThroughputLocalControlGroup(String groupName, int t
      * @return A {@link ThroughputControlGroup}.
      */
     @Beta(value = Beta.SinceVersion.V4_13_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
-    ThroughputControlGroup enableThroughputLocalControlGroup(String groupName, double targetThroughputThreshold) {
-        return this.enableThroughputLocalControlGroup(groupName, targetThroughputThreshold, false);
+    public ThroughputControlGroup enableThroughputLocalControlGroup(String groupName, double targetThroughputThreshold) {
+        return this.enableThroughputControlGroup(
+            ThroughputControlGroupFactory.createThroughputLocalControlGroup(groupName, this, targetThroughputThreshold));
     }
 
     /**
      *
      * @param groupName The throughput control group name.
      * @param targetThroughputThreshold The target throughput threshold for the control group.
      * @param isDefault Flag to indicate whether this group will be used as default.
+     *
+     * @return A {@link ThroughputControlGroup}.
+     */
+    @Beta(value = Beta.SinceVersion.V4_13_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public ThroughputControlGroup enableThroughputLocalControlGroup(String groupName, double targetThroughputThreshold, boolean isDefault) {
+        return this.enableThroughputControlGroup(
+            ThroughputControlGroupFactory.createThroughputLocalControlGroup(groupName, this, targetThroughputThreshold, isDefault));
+    }
+
+    //endregion
+
+    // region enableThroughputGlobalControlGroup
+
+    /**
+     *
+     * @param groupName The throughput control group name.
+     * @param targetThroughput The target throughput for the control group.
+     * @param globalControlContainer The global control container. This is the container which will be used to track the traffic load of each client.
+     * @param controlItemRenewInterval The global control container item renew interval.
+     * @param controlItemExpireInterval The global control container item expire interval.
+     *
      * @return A {@link ThroughputControlGroup}.
      */
     @Beta(value = Beta.SinceVersion.V4_13_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
-    ThroughputControlGroup enableThroughputLocalControlGroup(String groupName, double targetThroughputThreshold, boolean isDefault) {
-        return this.enableThroughputControlGroup(groupName, null, targetThroughputThreshold, ThroughputControlMode.LOCAL, isDefault);
+    public ThroughputControlGroup enableThroughputGlobalControlGroup(
```

Comment:
What about a model where you put these parameters into a wrapper object.

Like GlobalControlGroup { groupName, targetThroughput, container, renewal, expire.}

Then have CosmosClient/CosmosAsyncClient with createGlobalControlGroupBuilder(String dabaseId, String containerid)
That would basically pass the CosmosAsyncContainer down into the Builder
The other parameetrs can be set via the builder

build() returns the immutable control group.

CosmosAsyncCOntainer/Container have enableThrouhgputControlGroup(GlobalControlGroup)

NOTE - the Client used to create the Contorl group could even be a different client instance than for the container doing the enforcement. This way you avoid the heated discussions whether CosmosAsyncContainer should show up as parameter - and to be honest I would prefer the builder pattern vs. the large number of overloads for enableThroughputControlGroup as well

Reply:
Thanks Fabian, I like this proposal, I have changed to use builder pattern for both: global control and local control. And example: 

 ThroughputControlGroupConfig groupConfig =
            new ThroughputControlGroupConfigBuilder()
                .setGroupName("group-" + UUID.randomUUID())
                .setTargetThroughput(6)
                .build();

        ThroughputGlobalControlConfig globalControlConfig = this.client.createThroughputGlobalControlConfigBuilder(this.database.getId(), controlContainerId)
            .setControlItemRenewInterval(Duration.ofSeconds(5))
            .setControlItemExpireInterval(Duration.ofSeconds(10))
            .build();
        container.enableThroughputGlobalControlGroup(groupConfig, globalControlConfig);